spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/LocalApiKeyVerifier.java

```diff
@@ -0,0 +1,85 @@
+package eu.coatrack.proxy.security;
+
+/*-
+ * #%L
+ * coatrack-proxy
+ * %%
+ * Copyright (C) 2013 - 2021 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import eu.coatrack.api.ApiKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+/**
+ * This class provides a method to determine an API keys validity.
+ *
+ * @author Christoph Baier
+ */
+
+@Service
+public class LocalApiKeyVerifier {
+
+    private static final Logger log = LoggerFactory.getLogger(ApiKeyAuthTokenVerifier.class);
+
+    private final LocalApiKeyManager localApiKeyManager;
+
+    public LocalApiKeyVerifier(LocalApiKeyManager localApiKeyManager) {
+        this.localApiKeyManager = localApiKeyManager;
+    }
+
+    public boolean isApiKeyAuthorizedConsideringLocalApiKeyList(String apiKeyValue) {
+        log.debug("Begin checking if the API key with the value {} is valid using the local API key list.",
+                apiKeyValue);
+
+        ApiKey apiKey = localApiKeyManager.findApiKeyFromLocalApiKeyList(apiKeyValue);
+        if (apiKey == null) {
+            return false;
+        } else {
+            return isApiKeyValid(apiKey) && localApiKeyManager.wasLatestUpdateOfLocalApiKeyListWithinDeadline(apiKey);
```

Comment:
General comment: it seems like we first do a lot of work to get/extract/handle/check an apiKey, and in the end we do not accept it because of this deadline. Wouldn't it make more sense to check this "deadline" on a higher level _before_ doing all the api key handling work? Because: if the deadline is overdue, we do not even need to check the key.

Reply:
Done. The deadline condition is now checked before the validity check.